### ISSUE TITLE
you need to replace deprecated ".isAlive" per ".is_alive" in two places for python-3.9rc1 compatiblity

### ISSUE THREAD

**Issue by stonebig** [pyzo/pyzo#713]
I fail to start Pyzo, fisrt time I see this. any idea ?

> WPy64-3902\scripts>pyzo
> Traceback (most recent call last):
>   File "WPy64-3902\python-3.9.0rc1.amd64\lib\runpy.py", line 197, in _run_module_as_main
>     return _run_code(code, main_globals, None,
>   File "WPy64-3902\python-3.9.0rc1.amd64\lib\runpy.py", line 87, in _run_code
>     exec(code, run_globals)
>   File "WPy64-3902\python-3.9.0rc1.amd64\Scripts\pyzo.exe\__main__.py", line 4, in <module>
>   File "WPy64-3902\python-3.9.0rc1.amd64\lib\site-packages\pyzo\__init__.py", line 81, in <module>
>     if commandline.is_our_server_running():
>   File "WPy64-3902\python-3.9.0rc1.amd64\lib\site-packages\pyzo\core\commandline.py", line 150, in is_our_server_running
>     return server and server.isAlive()
> AttributeError: 'Server' object has no attribute 'isAlive'
> 
> WPy64-3902\scripts>

**Comment by stonebig** [comment]
package list:

````
Package                       Version
----------------------------- ----------------
adodbapi                      2.6.1.3
affine                        2.3.0
aiofiles                      0.5.0
aiohttp                       3.6.2
aiosqlite                     0.15.0
alabaster                     0.7.12
altair                        4.1.0
altair-data-server            0.4.1
altair-transform              0.2.0
altair-widgets                0.2.2
altgraph                      0.17
amply                         0.1.2
aniso8601                     8.0.0
ansiwrap                      0.8.4
appdirs                       1.4.4
argon2-cffi                   20.2.0.dev0
asciitree                     0.3.3
asgi-csrf                     0.6.1
asgiref                       3.2.10
asteval                       0.9.19
astroid                       2.4.2
astroML                       0.4.1
astropy                       4.0.1
async-generator               1.10
async-timeout                 3.0.1
atomicwrites                  1.4.0
attrs                         20.2.0
autopep8                      1.5.3
Babel                         2.8.0
backcall                      0.2.0
backports-abc                 0.5
baresql                       0.7.6
beautifulsoup4                4.9.1
black                         19.10b0
bleach                        3.1.5
blinker                       1.4
blosc                         1.9.1
bloscpack                     0.16.0
bokeh                         2.2.1
Bottleneck                    1.3.2
bqplot                        0.12.17
branca                        0.4.1
brewer2mpl                    1.4.1
Brotli                        1.0.9
cachelib                      0.1.1
Cartopy                       0.18.0
certifi                       2020.6.20
cffi                          1.14.2
cftime                        1.2.1
chardet                       3.0.4
click                         7.1.2
click-default-group           1.2.2
click-plugins                 1.1.1
cligj                         0.5.0
cloudpickle                   1.6.0
clrmagic                      0.0.1a2
colorama                      0.4.3
colorcet                      2.0.2
comtypes                      1.1.7
cvxopt                        1.2.5
cvxpy                         1.1.5
cx-Freeze                     6.2
cycler                        0.10.0
Cython                        0.29.21
cytoolz                       0.10.1
dask                          2.25.0
dask-glm                      0.2.0
dask-labextension             3.0.0
dask-ml                       1.6.0
dask-searchcv                 0.2.0
databases                     0.3.2
datasette                     0.48
datashader                    0.11.1
datashape                     0.5.2
db.py                         0.5.3
decorator                     4.4.2
defusedxml                    0.6.0
Deprecated                    1.2.10
descartes                     1.1.0
distlib                       0.3.1
distributed                   2.25.0
docopt                        0.6.2
docrepr                       0.1.1
docutils                      0.16
ecos                          2.0.7.post1
emcee                         3.0.2
entrypoints                   0.3
fast-histogram                0.9
fasteners                     0.15
fastparquet                   0.4.1
filelock                      3.0.12
Fiona                         1.8.17
Flask                         1.1.2
flask-accepts                 0.17.3
Flask-Mail                    0.9.1
flask-restx                   0.2.0
Flask-Session                 0.3.2
Flask-SQLAlchemy              2.4.4
flaskerize                    0.14.0
folium                        0.11.0
formlayout                    2.0.0a0
fs                            2.4.11
fsspec                        0.8.0
future                        0.18.2
fuzzywuzzy                    0.18.0
GDAL                          3.1.3
geographiclib                 1.50
geopandas                     0.8.1
geopy                         2.0.0
geoviews                      1.8.1
greenlet                      0.4.16
guidata                       1.7.9
guiqwt                        3.0.5
h11                           0.9.0
h2                            3.2.0
h5py                          2.10.0
HeapDict                      1.0.1
holoviews                     1.13.3
hpack                         3.0.0
html5lib                      1.1
httpie                        2.2.0
hupper                        1.10.2
husl                          4.0.3
hvplot                        0.6.0
Hypercorn                     0.10.2
hyperframe                    5.2.0
hypothesis                    5.29.4
idlex                         1.18
idna                          2.10
imageio                       2.9.0
imageio-ffmpeg                0.4.2
imagesize                     1.2.0
imbalanced-learn              0.7.0
importlib-metadata            1.7.0
iniconfig                     1.0.1
intake                        0.6.0
ipykernel                     5.3.4
ipyleaflet                    0.13.3
ipympl                        0.5.7
ipyparallel                   6.3.0
ipython                       7.18.1
ipython-genutils              0.2.0
ipython-sql                   0.4.0
ipywidgets                    7.5.1
isort                         5.5.1
itsdangerous                  1.1.0
janus                         0.5.0
jedi                          0.17.1
Jinja2                        2.11.2
joblib                        0.16.0
json5                         0.9.5
jsonschema                    3.2.0
julia                         0.5.4
jupyter                       1.0.0
jupyter-bokeh                 2.0.3
jupyter-client                6.1.7
jupyter-console               6.1.0
jupyter-core                  4.6.3
jupyter-server                0.1.1
jupyter-server-proxy          1.5.0
jupyter-sphinx                0.2.4
jupyterlab                    2.2.7
jupyterlab-launcher           0.13.1
jupyterlab-pygments           0.1.1
jupyterlab-server             1.2.0
keyring                       21.3.0
kiwisolver                    1.2.0
lazy-object-proxy             1.4.3
llvmlite                      0.34.0
lmfit                         1.0.1
locket                        0.2.0
loky                          2.8.0
lxml                          4.5.2
lz4                           3.1.0
Markdown                      3.2.2
MarkupSafe                    1.1.1
marshmallow                   3.7.1
matplotlib                    3.3.1
mccabe                        0.6.1
mercantile                    1.1.5
mergedeep                     1.3.0
metakernel                    0.27.0
mistune                       0.8.4
mizani                        0.7.1
mkl-service                   2.3.0
mlxtend                       0.17.3
monotonic                     1.5
more-itertools                8.5.0
moviepy                       1.0.3
mpl-scatter-density           0.7
mpld3                         0.5.1
mpldatacursor                 0.7.1
mpmath                        1.1.0
msgpack                       1.0.0
msvc-runtime                  14.27.29016
multidict                     4.7.6
multipledispatch              0.6.0
munch                         2.5.0
mypy                          0.782
mypy-extensions               0.4.3
mysql-connector-python        8.0.21
nbclient                      0.5.0
nbconvert                     5.6.1
nbconvert-reportlab           0.2
nbformat                      5.0.7
nest-asyncio                  1.4.0
netCDF4                       1.5.4
networkx                      2.4
NLopt                         2.6.2
nltk                          3.5
notebook                      6.1.4
numba                         0.51.2
numcodecs                     0.7.1
numdifftools                  0.9.39
numexpr                       2.7.2
numpy                         1.19.2+mkl
numpydoc                      1.1.0
oct2py                        5.2.0
octave-kernel                 0.32.0
osqp                          0.6.1
outcome                       1.0.1
packaging                     20.4
palettable                    3.3.0
pandas                        1.1.2
pandas-datareader             0.9.0
pandocfilters                 1.4.2
panel                         0.9.7
papermill                     2.1.3
param                         1.9.3
parambokeh                    0.2.3
paramnb                       2.0.4
parso                         0.7.0
partd                         1.1.0
passlib                       1.7.1
pathspec                      0.8.0
patsy                         0.5.1
pdfrw                         0.4
pdvega                        0.2.1.dev0
pefile                        2019.4.18
pep8                          1.7.1
pexpect                       4.8.0
pg8000                        1.13.1
pickleshare                   0.7.5
Pillow                        7.2.0
Pint                          0.15
pip                           20.2.3
pipdeptree                    1.0.0
pkginfo                       1.5.0.1
plotly                        4.9.0
plotnine                      0.7.1
pluggy                        0.13.1
ply                           3.11
portpicker                    1.3.1
ppci                          0.5.8
prettytable                   0.7.2
priority                      1.3.0
proglog                       0.1.9
prometheus-client             0.7.1
prompt-toolkit                3.0.7
protobuf                      4.0.0rc1
psutil                        5.7.2
ptpython                      3.0.5
ptvsd                         4.3.2
ptyprocess                    0.6.0
PuLP                          2.3
py                            1.9.0
pyaml                         20.4.0
PyAudio                       0.2.11
pybars3                       0.9.7
pybind11                      2.5.0
pycodestyle                   2.6.0
pycosat                       0.6.3
pycparser                     2.20
pyct                          0.4.6
pydeck                        0.4.0
pyepsg                        0.4.0
pyflakes                      2.2.0
pyflux                        0.4.17
pygame                        1.9.6
pygbm                         0.1.0
Pygments                      2.6.1
pyhdf                         0.10.2
PyInstaller                   3.6
pylint                        2.6.0
pymc                          2.3.8
PyMeta3                       0.5.1
pymongo                       3.11.0
pyodbc                        4.0.30
PyOpenGL                      3.1.5
pypandoc                      1.5
pyparsing                     2.4.7
pyproj                        2.6.1.post1
PyQt5                         5.15.1
PyQt5-sip                     12.8.1
pyqtgraph                     0.11.0
PyQtWebEngine                 5.15.1
pyrsistent                    0.17.0
pyserial                      3.4
pyshp                         2.1.0
pystache                      0.5.4
pytest                        6.0.1
python-baseconv               1.2.2
python-dateutil               2.8.1
python-dotenv                 0.14.0
python-hdf4                   0.10.0+dummy
python-jsonrpc-server         0.3.4
python-language-server        0.34.1
python-Levenshtein            0.12.0
python-snappy                 0.5.4
pythonnet                     2.5.1
PythonQwt                     0.8.3
pytz                          2020.1
pyviz-comms                   0.7.6
PyWavelets                    1.1.1
pywin32                       228
pywin32-ctypes                0.2.0
pywinpty                      0.5.7
pywinusb                      0.4.2
PyYAML                        5.3.1
pyzmq                         19.0.2
pyzo                          4.10.2
qtconsole                     4.7.7
QtPy                          1.9.0
quantecon                     0.4.8
Quart                         0.13.1
rasterio                      1.1.5
readme-renderer               26.0
redis                         3.5.3
regex                         2020.7.14
reportlab                     3.5.49
requests                      2.24.0
requests-toolbelt             0.9.1
retrying                      1.3.3
rfc3986                       1.4.0
rise                          5.6.1
rpy2                          3.4.0.dev0
Rtree                         0.9.4
ruamel.yaml                   0.16.10
Rx                            3.1.1
scikit-fuzzy                  0.4.1
scikit-image                  0.17.2
scikit-learn                  0.23.2
scikit-optimize               0.8.1
scilab2py                     0.6.2
scipy                         1.5.2
scs                           2.1.2
seaborn                       0.11.0
Send2Trash                    1.5.0
setuptools                    49.6.0
shap                          0.35.0
Shapely                       1.7.1
simpervisor                   0.3
simplegeneric                 0.8.1
simplejson                    3.17.2
simpy                         4.0.1
six                           1.15.0
snakeviz                      2.1.0
sniffio                       1.1.0
snowballstemmer               2.0.0
snuggs                        1.4.7
sortedcontainers              2.2.2
sounddevice                   0.4.0
soupsieve                     2.0.1
Sphinx                        3.2.1
sphinx-rtd-theme              0.5.0
sphinxcontrib-applehelp       1.0.2
sphinxcontrib-devhelp         1.0.2
sphinxcontrib-htmlhelp        1.0.3
sphinxcontrib-jsmath          1.0.1
sphinxcontrib-qthelp          1.0.3
sphinxcontrib-serializinghtml 1.1.4
SQLAlchemy                    1.3.19
sqlite-bro                    0.9.1
sqlparse                      0.3.1
statsmodels                   0.12.0
streamz                       0.5.4
supersmoother                 0.4
sympy                         1.6.2
tables                        3.6.1
tblib                         1.7.0
tenacity                      6.2.0
termcolor                     1.1.0
terminado                     0.8.3
testpath                      0.4.4
textwrap3                     0.9.2
threadpoolctl                 2.1.0
thrift                        0.13.0
tifffile                      2020.8.25
toml                          0.10.1
toolz                         0.10.0
tornado                       6.0.4
tqdm                          4.48.0
traitlets                     5.0.4
traittypes                    0.2.1
tranquilizer                  0.5.0+1.gff8dd5f
trio                          0.16.0
trio-asyncio                  0.11.0
twine                         3.2.0
twitter                       1.18.0
typed-ast                     1.4.1
typing-extensions             3.7.4.3
tzlocal                       2.1
umap-learn                    0.4.6
uncertainties                 3.1.4
urllib3                       1.25.10
uvicorn                       0.11.8
vega                          3.4.0
vega-datasets                 0.8.0
virtualenv                    20.0.31
ViTables                      3.0.2
voila                         0.1.23
wcwidth                       0.2.5
webencodings                  0.5.1
websockets                    8.1
Werkzeug                      1.0.1
wheel                         0.35.1
widgetsnbextension            3.5.1
winpython                     3.0.20200808
wordcloud                     1.8.0
wrapt                         1.12.1
wsproto                       0.15.0
xarray                        0.16.0
xlrd                          1.2.0
XlsxWriter                    1.3.3
xlwings                       0.20.2
yarl                          1.5.1
zarr                          2.4.0
zict                          2.0.0
zipp                          3.1.0

````

**Comment by stonebig** [comment]
I know I miss pynacl , paramiko, cryptography, not sure it's a problem.

**Comment by stonebig** [comment]
it seems to work on python-3.8 ... trying to digg further

**Comment by stonebig** [comment]
it looks like I have the same issue with Pyside2-5.15.1, ... so the problem is elsewhere ?

**Comment by stonebig** [comment]
ah ? "2019 answer: use is_alive instead of isAlive; you will see a deprecation warning for the above"

https://stackoverflow.com/questions/15063963/python-is-thread-still-running

**Comment by stonebig** [comment]
getting there

![image](https://user-images.githubusercontent.com/4312421/93019087-1822e280-f5d5-11ea-857c-8ad249fa12f6.png)

**Comment by stonebig** [comment]
missed there was a second isAlive:

````
  C:\WinP\bd39\bucod\WPy64-3902\python-3.9.0rc1.amd64\Lib\site-packages\pyzo\yoton\channels\channels_reqrep.py (1 hit)
	Line 703:             if not self._thread.isAlive():
````

**Comment by stonebig** [comment]
![image](https://user-images.githubusercontent.com/4312421/93019172-b616ad00-f5d5-11ea-8074-38f97caae03b.png)

**Comment by stonebig** [comment]
things to change (but don't know starting which python version)

> C:\WinP\a\pyzo-4.10.2\pyzo\core\commandline.py (1 hit)
> 	Line 150:     return server and server.isAlive()
>   C:\WinP\a\pyzo-4.10.2\pyzo\yoton\channels\channels_reqrep.py (1 hit)
> 	Line 703:             if not self._thread.isAlive():

**Comment by stonebig** [comment]
a long time ago deprecation:

````
isAlive()
Return whether the thread is alive.

This method returns True just before the run() method starts until just after the run() method terminates. The module function enumerate() returns a list of all alive threads.

Changed in version 2.6: Added is_alive() spelling.`
````

**Comment by almarklein** [comment]
Closed by #714
